src/main/java/frc/robot/subsystems/DriveSubsystem.java

```diff
@@ -11,103 +11,80 @@
 
+  private final DifferentialDriveKinematics kinematics = new DifferentialDriveKinematics(Units.inchesToMeters(26));
+
   private final boolean pidEnabled = false;
 
-  private final PIDController movePID;
-  private final PIDController turnPID;
+  /**
+   * @param ENC_TICKS_PER_FOOT The number of TalonFX encoder ticks per foot that
+   *                           the robot drives on 6" wheels. Used to input ft/s
+   *                           into WPI_TalonFX.set(VelocityControl, double
+   *                           value), which takes its value in encoder ticks per
+   *                           second.
+   * 
+   */
+  private final double ENC_TICKS_PER_FOOT = 1304;
 
   public DriveSubsystem() {
     // Slaves the left rear motor to the left front motor
     leftRear.follow(leftFront);
     rightRear.follow(rightFront);
-
-    // Initializes the 2 PIDControllers that will generate appropriate motor power
-    // values
-    movePID = new PIDController(1, 1, 1);
-    turnPID = new PIDController(1, 1, 1);
-
-    // Links both master-slave groups
-    drive = new DifferentialDrive(leftFront, rightFront);
-
-    drive.setDeadband(0);
   }
 
-  public void arcadeDrive(double moveRequest, double turnRequest, double speedLimiter) {
-    /*
-     * try(MoPerfMon.Period period =
-     * Robot.perfMon.newPeriod("DriveSubsystem::arcadeDrive")) {
-     */
-
-    // Scales the move and turn targets so they don't exceed the speed limit.
-    moveRequest *= speedLimiter;
-    turnRequest *= speedLimiter;
-
-    // Gets the actual forward/backward motion of the robot.
-    double moveRate = getMoveRate();
-    // Gets the acutal speed difference between left and right sides of the robot.
-    double turnRate = getTurnRate();
-
-    /**
-     * Generates a motor power value that will move the robot where the driver wants
-     * it to go, based on the difference between the current motor rate and the
-     * target.
-     */
-    double move = pidEnabled ? movePID.calculate(moveRate, moveRequest) : moveRequest;
-    double turn = pidEnabled ? turnPID.calculate(turnRate, turnRequest) : turnRequest;
-
-    // Forces the speed limit on the output of the PID controllers.
-    double m_r = Utils.clip(move, -speedLimiter, speedLimiter);
-    double t_r = Utils.clip(turn, -speedLimiter, speedLimiter);
-
-    // Sends the actual drive command that activates the motors.
-    drive.arcadeDrive(m_r, t_r, false);
-    // }
+  public void drive(double moveRequest, double turnRequest) {
+    var chassisSpeeds = new ChassisSpeeds(moveRequest, 0, turnRequest);
+    DifferentialDriveWheelSpeeds wheelSpeeds = kinematics.toWheelSpeeds(chassisSpeeds);
+    double leftMPerS = wheelSpeeds.leftMetersPerSecond;
+    double rightMPerS = wheelSpeeds.rightMetersPerSecond;
+    double leftETPerS = FeetToEncTicks(
+        Units.metersToFeet(leftMPerS) < 12 && Units.metersToFeet(leftMPerS) > -12 ? Units.metersToFeet(leftMPerS) : 12);
```

Comment:
1. You have a magic number. Document 12 with a constant
2. You are clipping leftMPerS between -12 and 12. You should look at the [Utils.clip()](https://github.com/momentumfrc/2020InfiniteRecharge/blob/master/src/main/java/frc/robot/utils/Utils.java#L4-L6) method
3. I was under the impression that 12ft/s was the maximum linear velocity of the robot (moveRequest), not the maximum velocity of the wheel. I'm not sure though, so that's a question for @terbos90803

Reply:
While I was writing this, I considered using clip(), but I wanted to get practice using ternary notation.  There's no other reason why I wrote out that logic.